Lab3.ipynb

```diff
@@ -0,0 +1,405 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 116,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sigma value: 5\n",
+      "Pos True: 100000\n",
+      "Pos False: 2\n",
+      "Neg True: 6\n",
+      "Neg False: 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "# 1.A.4: \n",
+    "sigma=5\n",
+    "goodKept = goodData[np.where((goodData< 12+sigma*.4) & (goodData> 12-sigma*.4))]\n",
+    "goodDiscard = len(goodData) - len(goodKept)\n",
+    "badData = np.asarray(badData)\n",
+    "badKept = badData[np.where((badData>12-sigma*.4) & (badData<12+sigma*.4))]\n",
+    "badDiscard = len(badData) - len(badKept)\n",
+    "print('Sigma value:', sigma)\n",
+    "print('Pos True:', len(goodKept))\n",
+    "print('Pos False:', len(badKept))\n",
+    "print('Neg True:', badDiscard)\n",
+    "print('Neg False:', goodDiscard)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 1.A.4: According to my truth table for a 5 sigma stastistical significance probability, we are keeping 2 bad data values and all the good data values. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 117,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sigma value: 1\n",
+      "Pos True: 68171\n",
+      "Pos False: 0\n",
+      "Neg True: 8\n",
+      "Neg False: 31829\n",
+      "\n",
+      "Sigma value: 2\n",
+      "Pos True: 95451\n",
+      "Pos False: 1\n",
+      "Neg True: 7\n",
+      "Neg False: 4549\n",
+      "\n",
+      "Sigma value: 3\n",
+      "Pos True: 99717\n",
+      "Pos False: 1\n",
+      "Neg True: 7\n",
+      "Neg False: 283\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# 1.B:\n",
+    "for i in range(1,4): \n",
+    "    goodKept = goodData[np.where((goodData< 12+i*.4) & (goodData> 12-i*.4))]\n",
+    "    goodDiscard = len(goodData) - len(goodKept)\n",
+    "    badData = np.asarray(badData)\n",
+    "    badKept = badData[np.where((badData>12-i*.4) & (badData<12+i*.4))]\n",
+    "    badDiscard = len(badData) - len(badKept)\n",
+    "    print('Sigma value:', i)\n",
+    "    print('Pos True:', len(goodKept))\n",
+    "    print('Pos False:', len(badKept))\n",
+    "    print('Neg True:', badDiscard)\n",
+    "    print('Neg False:', goodDiscard)\n",
+    "    print()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 1.B: Above I reran the code with different sigma values and as you can see the larger the threshhold the less likely you are to throw away good data. This is because more data will meet the demands of our threshold as the tolerancesa relax. And these omissions are predictable since we are dealing with a gaussian distribution. We know that within 1, 2, and 3 standard deviations the percentage of data inside that range is 68%, 95%, and 99.7% respectively."
```

Comment:
Problem 1.B:
- output tables look good and make sense, comparing with different sigma values is good
- typo in explanation: tolerancesa --> tolerances
- explanation about omission mistakes makes sense. Since there 100k points though, even at 99.7% (3 sigma) several good data points are still counted as bad ones. Consider also printing for sigma=4 since that is one of the closest sigma values for the statistical test.
- now that I'm looking closer, the labels for the output lines might be swapped. It took me awhile to understand what the labels meant, but I read them (based on the table provided in the lab instructions) as: "Good temperature data testing as good data", "bad temperature data testing as bad data" for the first diagonal. These are when the statistical test is correct. The other diagonal, I read as "Good temperature data testing as bad data" and "Bad temperature data testing as good data" , the failures of the statistical test (the omission and commission mistakes). 
So when I'm looking at the labels of your output I'm confused at what "negative false" is, because I think when the threshold is low (like for the sigma=1 truth table) that should be "Good temperature data testing as bad data" which in the lab instructions, is labelled as "True T / Your Test Bad". All the numbers look good, but I think putting it in table format helps because then you can see the diagonals of the table more clearly, and then you can see if the labels are swapped. 

(Labelling it as true and false confused me a lot and is something I should also probably fix in my own code, because it tripped me up as well. I think the labels given in the lab instructions are more appropriate now that I understand the truth tables after finishing the last HW assignment)